festim/hydrogen_transport_problem.py

```diff
@@ -32,11 +32,16 @@ def __init__(
         self.solver_parameters = solver_parameters
         self.exports = exports
 
+        self.dx = None
+        self.ds = None
+        self.function_spaces = []
```

Comment:
1 model = 1 function space

I don't see why you'd have several.

Proof is, in `self.functions_spaces` is only appended to in `initialise(),` which is a method only called once per simulation.

`self.function_spaces` shouldn't be a list.